flake8_trio/base.py

```diff
@@ -0,0 +1,54 @@
+"""Contains base classes used across multiple parts of the package."""
+
+from __future__ import annotations
+
+from typing import Any, NamedTuple
+
+
+class Statement(NamedTuple):
+    name: str
+    lineno: int
+    col_offset: int = -1
+
+    def __eq__(self, other: Any) -> bool:
+        return (
+            isinstance(other, Statement)
+            and self[:2] == other[:2]  # type: ignore
+            and (
+                self.col_offset == other.col_offset
+                or -1 in (self.col_offset, other.col_offset)
+            )
+        )
+
+
+class Error:
+    def __init__(
+        self, error_code: str, lineno: int, col: int, message: str, *args: object
+    ):
+        self.line = lineno
+        self.col = col
+        self.code = error_code
+        self.message = message
+        self.args = args
+
+    # for yielding to flake8
+    def __iter__(self):
+        yield self.line
+        yield self.col
+        yield f"{self.code} " + self.message.format(*self.args)
+        yield None
```

Comment:
This line (39) previous read `yield type(Plugin)`, but due to file moving/splitting the diff isn't picked up.

To avoid a circular import, and placing Plugin and error in the same file would also be problematic, I stop adhering to an outdated requirement that flake8 plugins should yield the type of their Plugin. I'm not finding it in their documentation now, but I remember the video tutorial mentioning that this parameter doesn't actually matter.